addon/models/course.js

```diff
@@ -1,271 +1,273 @@
+  @use _programs = new ResolveAsyncValue(() => [this._programYears?.mapBy('program')]);
+  @use _programSchools = new ResolveAsyncValue(() => [this._programs?.mapBy('school')]);
+  @use _resolvedSchool = new ResolveAsyncValue(() => [this.school]);
+  get schools() {
+    if (!this._programSchools || !this._resolvedSchool) {
+      return undefined;
+    }
+
+    return [...this._programSchools, this._resolvedSchool].uniq();
+  }
+
+  @use _schoolVocabularies = new ResolveAsyncValue(() => [this.schools?.mapBy('vocabularies')]);
+
+  get assignableVocabularies() {
+    return this._schoolVocabularies
+      ?.reduce((acc, curr) => {
+        return acc.pushObjects(curr.toArray());
       }, [])
       .sortBy('school.title', 'title');
-  }),
+  }
 
+  @use _courseObjectives = new ResolveAsyncValue(() => [this.courseObjectives]);
   /**
    * A list of course objectives, sorted by position (asc) and then id (desc).
-   * @property sortedCourseObjectives
-   * @type {Ember.computed}
    */
-  sortedCourseObjectives: computed('courseObjectives.@each.position', async function () {
-    const objectives = await this.courseObjectives;
-    return objectives.toArray().sort(sortableByPosition);
-  }),
-
-  hasMultipleCohorts: computed('cohorts.[]', function () {
-    const meta = this.hasMany('cohorts');
-    const ids = meta.ids();
+  get sortedCourseObjectives() {
+    return this._courseObjectives?.toArray().sort(sortableByPosition);
+  }
 
-    return ids.length > 1;
-  }),
+  get hasMultipleCohorts() {
+    return this.cohorts.length > 1;
+  }
 
+  @use _allTermVocabularies = new ResolveFlatMapBy(() => [this.terms, 'vocabulary']);
   /**
    * A list of all vocabularies that are associated via terms.
-   * @property associatedVocabularies
-   * @type {Ember.computed}
-   * @public
    */
-  associatedVocabularies: computed('terms.@each.vocabulary', async function () {
-    const terms = await this.terms;
-    const vocabularies = await all(terms.toArray().mapBy('vocabulary'));
-    return vocabularies.uniq().sortBy('title');
-  }),
+  get associatedVocabularies() {
+    return this._allTermVocabularies?.uniq().sortBy('title');
+  }
+
+  @use _allTermParents = new ResolveFlatMapBy(() => [this.terms, 'allParents']);
+  @use _resolvedTerms = new ResolveAsyncValue(() => [this.terms]);
 
   /**
    * A list containing all associated terms and their parent terms.
-   * @property termsWithAllParents
-   * @type {Ember.computed}
-   * @public
    */
-  termsWithAllParents: computed('terms.[]', async function () {
-    const terms = await this.terms;
-    const allTerms = await all(terms.toArray().mapBy('termWithAllParents'));
-    return allTerms
-      .reduce((array, set) => {
-        array.pushObjects(set);
-        return array;
-      }, [])
-      .uniq();
-  }),
+  get termsWithAllParents() {
+    if (!this._allTermParents || !this._resolvedTerms) {
+      return undefined;
```

Comment:
see comment about the `schools` getter above - possible to return a blank array here instead?

Reply:
✅